java/drivers/driver-hazelcast4plus/src/main/java/com/hazelcast/simulator/tests/vector/VectorCollectionSearchDatasetTest.java

```diff
@@ -117,19 +128,28 @@ public void setup() {
                                         .setDimension(dimension)
                                         .setMaxDegree(maxDegree)
                                         .setEfConstruction(efConstruction)
+                                        .setUseDeduplication(useDeduplication)
                         )
         );
 
-        options = new SearchOptionsBuilder()
-                .includeValue()
-                .includeVectors()
-                .limit(limit)
-                .build();
+        SearchOptionsBuilder optionsBuilder = SearchOptions.builder()
+                .setIncludeValue(includeValue)
+                .setIncludeVectors(includeVectors)
+                .limit(limit);
+        if (efSearch != null) {
+            optionsBuilder.hint(Hints.EF_SEARCH, efSearch);
+        }
+        if (singleStage) {
+            optionsBuilder.hint(Hints.FORCE_SINGLE_STAGE_SEARCH, true);
+        }
+        options = optionsBuilder.build();
     }
 
     @Prepare(global = true)
     public void prepare() {
-        var size = Math.min(reader.getSize(), loadFirst);
```

Comment:
loadFirst was quite a convenient parameter for debugging. May it be worth to preserve it.a

Reply:
I replaced it with a more general `targetCollectionSize`